### ISSUE TITLE
False positive on S4039: Interface and base class should be satisfied

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2757]
### Description
Explicit implementations of an interface are sometimes the only way deal with conflicting contracts:

### Repro steps
``` C#
public class S4039 : S4039Double, I4039Int
{
    public S4039(int value) : base(value) { }

    int I4039Int.Value => (int)Value; // compliant

    bool I4039Int.Add(int val) // compliant
    {
        Add(val);
        return true;
    }
}

public class S4039Double
{
    public S4039Double(double val) => Value = val;

    public double Value { get; private set; }
    public void Add(int val) => Value += val;
}
public interface I4039Int
{
    int Value { get; }
    bool Add(int val);
}
```

### Expected behavior
Although not desired, there is not way to deal with this problem without using explicit declarations of the interface, as the base class and the implemented interface disagree on the type of the `Value` property, or the the return type of the `Add` method.

### Related information

* SonarC# Version: 8.0.0

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1978](https://sonarsource.atlassian.net/browse/NET-1978)

[NET-1978]: https://sonarsource.atlassian.net/browse/NET-1978?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
